src/CommunityToolkit.Maui/Layouts/StateLayout/StateLayoutController.shared.cs

```diff
@@ -0,0 +1,285 @@
+				// StateView to allow for more control over how it layouts.
+				var s = new VerticalStackLayout
+				{
+					Opacity = animate ? 0 : 1,
+					VerticalOptions = view.VerticalOptions,
+					HorizontalOptions = view.HorizontalOptions
+				};
+
+				// If the layout we're applying StateLayout to is a Grid,
+				// we want to have the StateLayout span the entire Grid surface.
+				// Otherwise it would just end up in row 0 : column 0.
+				if (layout is Grid grid)
+				{
+					if (grid.RowDefinitions.Count > 0)
+					{
+						Grid.SetRowSpan(s, grid.RowDefinitions.Count);
+					}
+
+					if (grid.ColumnDefinitions.Count > 0)
+					{
+						Grid.SetColumnSpan(s, grid.ColumnDefinitions.Count);
+					}
+
+
+					layout.Children.Add(s);
+					layoutIsGrid = true;
+				}
+
+				var itemView = CreateItemView(state, customState);
+
+				if (itemView is not null)
+				{
+					if (layoutIsGrid)
+					{
+						s.Children.Add(itemView);
+					}
+					else
+					{
+						layout.Children.Add(itemView);
+					}
+
+				}
+			}
+
+			await FadeLayoutChildren(layout, animate, false, token);
+		}
+	}
+
+	internal Layout GetLayout()
+	{
+		layoutWeakReference.TryGetTarget(out var layout);
+		return layout ?? throw new ObjectDisposedException("Layout Disposed");
+	}
+
+	static async ValueTask FadeLayoutChildren(Layout layout, bool shouldAnimate, bool isHidden, CancellationToken token)
+	{
+		if (shouldAnimate && layout.Children.Count > 0)
+		{
+			var opacity = 1;
+			var time = 500u;
+
+			if (isHidden)
+			{
+				opacity = 0;
+				time = 100u;
+			}
+
+			await Task.WhenAll(layout.Children.OfType<View>().Select(a => ViewExtensions.FadeTo(a, opacity, time))).WaitAsync(token);
+		}
+	}
+
+	StateView? GetViewForState(LayoutState state, string? customState)
+	{
+		var view = StateViews.FirstOrDefault(x => (x.StateKey == state && state is not LayoutState.Custom) ||
+						(state is LayoutState.Custom && x.CustomStateKey == customState));
+
+		return view;
+	}
+
+	int GetRepeatCount(LayoutState state, string? customState)
+	{
+		var template = StateViews.FirstOrDefault(x => (x.StateKey == state && state is not LayoutState.Custom) ||
+					   (state is LayoutState.Custom && x.CustomStateKey == customState));
+
+		return template?.RepeatCount ?? 1;
+	}
+
+	DataTemplate? GetTemplate(LayoutState state, string? customState)
+	{
+		var view = StateViews.FirstOrDefault(x => (x.StateKey == state && state is not LayoutState.Custom) ||
+					   (state is LayoutState.Custom && x.CustomStateKey == customState));
+
+		return view?.Template;
+	}
+
+	View CreateItemView(LayoutState state, string? customState)
+	{
+		var view = StateViews.FirstOrDefault(x => (x.StateKey == state && state is not LayoutState.Custom) ||
+						(state is LayoutState.Custom && x.CustomStateKey == customState));
+
+		// TODO: This only allows for a repeatcount of 1.
+		// Internally in Xamarin.Forms we cannot add the same element to Children multiple times.
```

Comment:
I'm guessing this is a copy/pasted comment from the Xamarin.CommunityToolkit implementation. 

Can we do this ToDo before merging this PR?

Reply:
It seems the author ran into the same limitation cloning elements as I did - see RepeatCount discussion below. ToDO is removed for now.